pelicun/model/loss_model.py

```diff
@@ -850,20 +847,44 @@ def _validate_input_loss_combination(self, loss_combination: dict) -> None:
                     )
 
     def _validate_input_replacement_thresholds(
-        self, replacement_thresholds: dict
+        self,
+        replacement_configuration: tuple[
+            uq.RandomVariableRegistry, dict[str, float]
+        ],
     ) -> None:
-        for key, _value in replacement_thresholds.items():
+        replacement_consequence_RV_reg, replacement_ratios = (
+            replacement_configuration
+        )
+        if not isinstance(replacement_consequence_RV_reg, uq.RandomVariableRegistry):
+            raise TypeError(
+                f'Invalid type for replacement consequence RV registry: '
+                f'{type(replacement_consequence_RV_reg)}. It should be '
+                f'uq.RandomVariableRegistry.'
+            )
+        for key in replacement_consequence_RV_reg.RV:
+            if key not in self.decision_variables:
+                raise ValueError(
+                    f'`replacement_consequence_RV_reg` contains the key '
+                    f'`{key}` which is not found in the active '
+                    f'decision variables. These are: '
+                    f'{self.decision_variables}.'
```

Comment:
My view is that the approach you describe encourages the behavior of expecting to get and ignore warnings, which in turn might result in users not even looking at the warnings after some point. I think we should aim for our warnings to be taken seriously and assist the users in identifying unintentional issues with their inputs. So I would suggest adding an option to remove DVs from `replacement_consequence_RV_reg` not present in `self.decision_variables`.

Reply:
Do you suggest adding this to the list of Options? And, for the users I described above, this would be set to True and when doing the checks in the referenced code, you would remove the DVs rather than throwing an error, right?

I think this is a great idea.

Reply:
Added with 544d0c75.
It is unnecesasry to remove the extra DVs from the RV registry since they are not accessed.
544d0c75 is the first to contain a configuration for how errors are handled. I defined a dedicated section in the default options JSON file and I'm accessing it with a new assessment object method to better handle nesting. We can follow the same pattern for other similar options.